json-diff.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * json-diff.js - JSON file diff.
+ *
+ * Copyright © 2020, JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var fs = require('fs');
+var util = require('util');
+var path = require("path");
+
+function usage() {
+    console.log("Usage: json-diff [file1] [file2]\n" +
+        "diff.js\n" +
+        "  It shows the result of adding and deleting to file2 compared to file1.\n" +
+        "-h or --help\n" +
+        "file1\n" +
+        "  json file1.\n" +
+        "file2\n" +
+        "  json file2.\n");
+    process.exit(1);
+}
+
+process.argv.forEach(function (val, index, array) {
+    if (val === "-h" || val === "--help") {
+        usage();
+    }
+});
+
+var jsonOrg = process.argv[2];
+var jsonNew = process.argv[3];
+
+var resourceObj, resourceObjNew;
+var newString = [], removedString = [];
+
+function readJSONs(jsonOrg, jsonNew) {
+    if (!fs.existsSync(jsonOrg)) {
+        util.error("Could not access file: " + jsonOrg);
+        usage();
+    } else {
+        resourceObj = require(path.resolve(jsonOrg));
+    }
+
+    if (!fs.existsSync(jsonNew)) {
+        util.error("Could not access file: " + jsonNew);
+        usage();
+    } else {
+        resourceNewObj = require(path.resolve(jsonNew));
+    }
+}
+
+function compareFile(resourceObj, resourceNewObj) {
+    for (var item in resourceObj) {
+        if (!resourceNewObj.hasOwnProperty(item)) {
+            removedString.push(item);
```

Comment:
This only compares the property names, not the property values.  Also, it is not recursive, so it only does a shallow compare. 

For a deep compare, see this function: https://github.com/iLib-js/assertExtras/blob/master/assertExtras.js#L101
Copy that here, and you can do a very fine grain diff.

Reply:
Before modifying codes I want to make sure,  There are some that I assumed for this script.
1. This script is for comparing the property name  Because  I have interested only if I don't miss any strings during a loctool's extracting strings process. If the property names haven't changed, It doesn't matter the values are changed or not.
2. I assumed the JSON resource format would have below form all the time which doesn't have depth.
`{"a": "hello", "b": "hi"} `

Would it be worth that updating code for future usage?